core/src/main/java/org/mule/runtime/core/internal/processor/strategy/WorkQueueStreamProcessingStrategyFactory.java

```diff
@@ -87,6 +117,43 @@ protected WorkQueueStreamProcessingStrategy(Supplier<Scheduler> ringBufferSchedu
            maxConcurrencyEagerCheck, false);
     }
 
+    @Override
+    public Sink createSink(FlowConstruct flowConstruct, ReactiveProcessor function) {
+      final long shutdownTimeout = flowConstruct.getMuleContext().getConfiguration().getShutdownTimeout();
+      WorkQueueProcessor<EventWrapper> processor =
+          WorkQueueProcessor.<EventWrapper>builder().executor(ringBufferSchedulerSupplier.get()).bufferSize(bufferSize)
+              .waitStrategy(waitStrategy.getReactorWaitStrategy()).build();
+      int subscriberCount = maxConcurrency < subscribers ? maxConcurrency : subscribers;
+      CountDownLatch completionLatch = new CountDownLatch(subscriberCount);
+      for (int i = 0; i < subscriberCount; i++) {
+        processor
+            .doOnSubscribe(subscription -> currentThread().setContextClassLoader(executionClassloader))
+            .map(ew -> ew.getWrappedEvent())
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this lambda with a method reference. ([squid:S1612](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1612))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7708&issues=AWkQk2wOotIXnmsUMK6V&open=AWkQk2wOotIXnmsUMK6V)

[ISSUE_KEY]: AWkQk2wOotIXnmsUMK6V